js-packages/@prestojs/ui/src/getFormatterForField.ts

```diff
@@ -99,11 +97,14 @@ export default function getFormatterForField<
     ): React.ComponentType<T> | [React.ComponentType<T>, Record<string, unknown>] | string => {
         if (f.choices) {
             if (Array.isArray(w)) {
-                return [w[0], { ...w[1], choices: f.choices }];
+                return [w[0], { ...w[1], choices: f.choices, ...f.getFormatterProps() }];
             } else {
-                return [w, { choices: f.choices }];
+                return [w, { choices: f.choices, ...f.getFormatterProps() }];
             }
         } else {
+            if (w) {
```

Comment:
hmm, when would w be falsy? it looks like its also assumed w would exist if f is a choices

Reply:
hmm yeah looks like you are right - i'll add missing types to make it more obvious